### PR TITLE
[CP][Android] Revert "Reland: "Fix how Gradle resolves Android plugin" (#137115)"

### PR DESCRIPTION
**Pull request by camsim99**
Cherry-pick for https://github.com/flutter/flutter/pull/137115 to fix https://github.com/flutter/flutter/issues/141940.  Steps taken to create this cherry-pick:

1. Cherry pick commit 995e3fa and fix merge conflicts
2. Delete the following code in `flutter.groovy` that I accidentally kept in the merge but was added in https://github.com/flutter/flutter/pull/137115
```groovy
// Load shared gradle functions
project.apply from: Paths.get(flutterRoot.absolutePath, "packages", "flutter_tools", "gradle", "src", "main", "groovy", "native_plugin_loader.groovy")
```
3. Deleted `android_plugin_skip_unsupported_test.dart` that was left incorrectly by merge.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/test_data/plugin_project.dart
Comment (reidbaker) [inline packages/flutter_tools/test/integration.shard/test_data/plugin_project.dart:1]: Should this file be deleted? Same with packages/flutter_tools/test/integration.shard/test_data/plugin_each_settings_gradle_project.dart

Comment (camsim99) [reply packages/flutter_tools/test/integration.shard/test_data/plugin_project.dart:1]: These were both added (along with packages/flutter_tools/test/integration.shard/android_plugin_skip_unsupported_test.dart) in https://github.com/flutter/flutter/pull/137115

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -392,77 +390,19 @@ class FlutterPlugin implements Plugin<Project> {
...
+    private void configurePlugins() {
+        getPluginList().each(this.&configurePluginProject)
+        getPluginDependencies().each(this.&configurePluginDependencies)
     }
 
     /** Adds the plugin project dependency to the app project. */
-    private void configurePluginProject(Map<String, Object> pluginObject) {
-        assert(pluginObject.name instanceof String)
-        Project pluginProject = project.rootProject.findProject(":${pluginObject.name}")
+    private void configurePluginProject(String pluginName, String _) {
```
Comment (camsim99) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:402]: As per a recommendation from @reidbaker, I downloaded a Groovy extension in VSCode to see if there were any issues and the only relevant one to lines modified here are lint warnings that the `String _` variable should be named and that it's not referenced in the method. However, this was how it was before until it was changed in the [original PR](https://github.com/flutter/flutter/pull/137115/files#diff-f0cdaf8963c97a175f64fe0fad130260d0614dd6cecd5cbe9580628bd51c5a4cL402), so should be fine!

FWIW the other errors/warnings are also lints.

Comment (reidbaker) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:402]: > As per a recommendation from @reidbaker, I downloaded a Groovy extension in VSCode to see if there were any issues and the only relevant one to lines modified here are lint warnings that the `String _` variable should be named and that it's not referenced in the method. However, this was how it was before until it was changed in the [original PR](https://github.com/flutter/flutter/pull/137115/files#diff-f0cdaf8963c97a175f64fe0fad130260d0614dd6cecd5cbe9580628bd51c5a4cL402), so should be fine!
> 
> FWIW the other errors/warnings are also lints.

 Thanks for checking this was the last thing I could think of to help us gain confidence in the revert not creating more issues.

**Review by reidbaker — APPROVED** [review]

**Review by bartekpacia — APPROVED** [review]

**Comment by bartekpacia** [comment]
Can I submit "Add support for Gradle Kotlin DSL" again now?

**Comment by bartekpacia** [comment]
Yes🙌

**Pull request merged**